boa/src/builtins/function/mod.rs

```diff
@@ -520,11 +534,19 @@ impl BuiltInFunctionObject {
                 Some(name),
             ) => Ok(format!("function {}() {{\n  [native Code]\n}}", &name).into()),
             (Function::Ordinary { body, params, .. }, Some(name)) => {
-                let arguments: String = params
-                    .iter()
-                    .map(|param| param.name())
-                    .collect::<Vec<&str>>()
-                    .join(", ");
+                let arguments: String = {
+                    let mut argument_list: Vec<Cow<'_, str>> = Vec::new();
+                    for params_item in params.iter() {
+                        let argument_item = match &params_item.declaration() {
+                            Declaration::Identifier { ident, .. } => Cow::Borrowed(ident.as_ref()),
+                            Declaration::Pattern(pattern) => {
+                                Cow::Owned(format!("{{{}}}", pattern.idents().join(",")))
```

Comment:
Why are there triple nested braces inside the `format!`? Is it necessary or is just a typo?

Reply:
the inner most one is for position and the other two are used to include "{}" in the string, "{{}}" is used to signify "{}" in format!

Reply:
I see, pretty cool then :)